cdc/sorter/unified/unified_sorter.go

```diff
@@ -281,6 +281,23 @@ func (s *Sorter) AddEntry(ctx context.Context, entry *model.PolymorphicEvent) {
 	}
 }
 
+// TryAddEntry implements the EventSorter interface
+func (s *Sorter) TryAddEntry(ctx context.Context, entry *model.PolymorphicEvent) bool {
+	select {
+	case <-ctx.Done():
+		return false
+	case <-s.closeCh:
+		return false
+	default:
+	}
+	select {
```

Comment:
```suggestion
```

Could you merge two select?

Reply:
add two select to guarantee the done/close condition is checked first.

Reply:
I see, could you add some some comments?

Reply:
done